develop/CICD/Tutorials/CICD_For_Other_Items/CICD_Tutorial_For_Other_Items_User_Defined_API_VisualStudio_And_GitHub.md

```diff
@@ -0,0 +1,105 @@
+---
+uid: CICD_Tutorial_For_Other_Items_User_Defined_API_VisualStudio_And_GitHub
+---
+
+# Registering a new version of a user-defined API to the Catalog using Visual Studio and GitHub
+
+In this tutorial, you will learn how to develop, (pre-)release, and upload a user-defined API to the DataMiner Catalog with a CI/CD pipeline using Visual Studio and GitHub. However, this tutorial can be used for **any artifact type** except for connectors (e.g. Automation script, Automation script library, package project, etc.).
+
+Expected duration: 10 minutes
+
+## Prerequisites
+
+- A [GitHub account](https://docs.github.com/en/get-started/signing-up-for-github/signing-up-for-a-new-github-account)
+- [DataMiner Integration Studio](https://community.dataminer.services/exphub-dis/)
+- [Microsoft Visual Studio 2022](https://visualstudio.microsoft.com/downloads/)
+- A dataminer.services account with the *Owner* role in order to be able to access/create keys for your organization.
+
+## Overview
+
+- [Step 1: Create a Visual Studio project](#step-1-create-a-visual-studio-project)
+- [Step 2: Create a GitHub repository](#step-2-create-a-github-repository)
+- [Step 3: Check the GitHub Actions](#step-3-check-the-github-actions)
+- [Step 4: Create and add a secret](#step-4-create-and-add-a-secret)
+- [Step 5: Check the results](#step-5-check-the-results)
+
+## Step 1: Create a Visual Studio project
+
+1. Open Visual Studio, and select *Create a new project*.
+
+1. Select the *DataMiner User-Defined API Project* template, and click *Next*.
+
+1. Enter a project name, for example *MyUserDefinedApi*.
+
+1. Make sure *Place solution and project in the same directory* is **not selected**, and click *Next*.
+
+1. Fill in your name as the author and keep *Create DataMiner Package* selected.
+
+1. Under *Add GitHub CI/CD Workflow (Overwrite Existing)*, select *Basic*.
+
+1. Click *Create*.
+
+## Step 2: Create a GitHub repository
+
+1. In the menu bar of Visual Studio, select *GIT* > *Create GIT Repository*.
+
+1. Create a new GitHub repository:
+
+   1. Select a name, specify your GitHub account, and mark yourself as the owner.
+
+   1. When you have entered all the information, click *Create and Push*.
+
+1. Go to the newly created GitHub repository on <https://github.com/>.
+
+   > [!TIP]
+   > In Visual Studio 2022, you can open the *GIT* menu and select *GitHub/View Pull Requests* to quickly access the correct repository.
+
+## Step 3: Check the GitHub Actions
+
+1. Create a GitHub repository by going to **Git > Create Git Repository**, selecting GitHub, and filling in the wizard before clicking **Create and Push**.
```

Comment:
Same question as before: Is it correct that you need to create a GitHub repo again as part of this step (even though this was already done in step 2), or should this be removed here?

Reply:
Please remove. same as above.